src/components/Login/Login.jsx

```diff
@@ -0,0 +1,130 @@
+import React, { useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useHistory } from 'react-router';
+import { Button, Form, Spinner } from 'react-bootstrap';
+import { useCookies } from 'react-cookie';
+import axios from 'axios';
+
+import {
+    loginPending,
+    loginSuccessful,
+    loginFail,
+} from '../../redux/authSlice';
+import { setMessage } from '../../redux/messageAlertSlice';
+
+let Login = () => {
+    const [userName, setUserName] = useState('');
+    const [userPassword, setUserPassword] = useState('');
+    const [rememberMe, setRememberMe] = useState(false);
+    const [_, setCookie] = useCookies(['authToken']);
+    const [networkError, setNetworkError] = useState(false);
+    const history = useHistory();
+    const { isLoading, error } = useSelector((state) => state.auth);
+    const dispatch = useDispatch();
+    function loginUser(e) {
+        e.preventDefault();
+        dispatch(loginPending());
+        const userLoginCreds = {
```

Comment:
please try to not to keep the logic in react component and try to keep the presentation layer in the react component. so better to create  a library, package or js file in you app for user authentication & registration. which should have the utilities related to login, forget password, signup etc.